docs/src/main/asciidoc/datasource-guide.adoc

```diff
@@ -0,0 +1,212 @@
+* the Maven structure
+* a landing page accessible on `http://localhost:8080`
+* example `Dockerfile` files for both `native` and `jvm` modes
+* the application configuration file
+* an `org.acme.events.GreetingResource` resource
+* an associated test
+
+== Adding maven dependencies
+
+Next, you will need to add the `quarkus-agroal` dependency to your project.
+There are a couple of ways to do this.
+You can add it explicitly, or transitively by adding the `quarkus-hibernate-orm` dependency.
+
+If you want to add the dependency explicitly:
+
+[source,xml]
+--
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-agroal</artifactId>
+</dependency>
+--
+
+If you'd like to add Hibernate ORM instead, and get Agroal transitively:
+
+[source,xml]
+--
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-hibernate-orm</artifactId>
+</dependency>
+--
+
+You will also need to add the database connector library of choice.
+Quarkus supports
+
+* H2
+* MariaDB (and MySQL)
+* Postgresql
+
+Do so by adding one of these dependencies to your pom.xml file:
+
+[source,xml]
+--
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-jdbc-h2</artifactId>
+</dependency>
+--
+[source,xml]
+--
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-jdbc-mariadb</artifactId>
+</dependency>
+--
+[source,xml]
+--
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-jdbc-postgresql</artifactId>
+</dependency>
+--
+
+== Configuring the datasource
+
+After all of the dependencies are added to your pom.xml file, you'll need to configure Agroal.
+This is done in `{config-file}`.
+The minimum necessary for a configured database is as follows:
+
+[source,properties]
+--
+include::../../../../extensions/agroal/deployment/src/test/resources/application-named-datasource.properties[]
+--
+
+There are other configuration options, detailed below.
+
+== Injecting a Datasource
+
+Because {project-name} uses CDI, injecting a Datasource is very simple:
+
+[source,java,indent=0]
+--
+include::../../../../extensions/agroal/deployment/src/test/java/io/quarkus/agroal/test/DefaultDataSourceConfigTest.java[tag=injection]
+--
+
+In the above example, the type is `AgroalDataSource` which is a subtype of `javax.sql.DataSource`.
+Because of this, you can also use `javax.sql.DataSource`.
+
+== Multiple Datasources
+
+Agroal allows you configure multiple datasources.
+It works exactly the same way as a single datasource, with one important change: a name.
+
+[source,properties]
+--
+include::../../../../extensions/agroal/deployment/src/test/resources/application-multiple-datasources.properties[]
+--
+
+Notice there's an extra bit in the key.
+The syntax is as follows: `quarkus.datasource.[optional name.][datasource property]`.
+
+=== Named Datasource Injection
+
+When using multiple datasources, each `DataSource` also has the `Named` qualifier with the name of the datasource in the property as the value.
```

Comment:
I thought we decided against that and we decided to keep the `@DataSource` qualifier as it was more CDI?